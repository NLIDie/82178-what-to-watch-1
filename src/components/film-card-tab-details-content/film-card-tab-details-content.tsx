import * as React from 'react';

// Types
import { TProps } from './film-card-tab-details-content.types';

export function FilmCardTabDetailsContent(props: TProps): JSX.Element {
  const {
    director,
    starring,
    runTime,
    genre,
    released
  } = props;

  return (
    <React.Fragment>
      <div className="movie-card__text-col">
        <p className="movie-card__details-item">
          <strong className="movie-card__details-name">
            Director
          </strong>
          <span className="movie-card__details-value">
            {director}
          </span>
        </p>
        <p className="movie-card__details-item">
          <strong className="movie-card__details-name">
            Starring
          </strong>

          <span className="movie-card__details-value">
            {starring.map((star: string) => (
              <span key={star}>
                {star}
                <br />
              </span>
            ))}
          </span>
        </p>
      </div>

      <div className="movie-card__text-col">
        <p className="movie-card__details-item">
          <strong className="movie-card__details-name">
            Run Time
          </strong>
          <span className="movie-card__details-value">
            {runTime}
          </span>
        </p>

        <p className="movie-card__details-item">
          <strong className="movie-card__details-name">Genre</strong>
          <span className="movie-card__details-value">
            {genre}
          </span>
        </p>

        <p className="movie-card__details-item">
          <strong className="movie-card__details-name">
            Released
          </strong>
          <span className="movie-card__details-value">
            {released}
          </span>
        </p>
      </div>
    </React.Fragment>
  )
}
