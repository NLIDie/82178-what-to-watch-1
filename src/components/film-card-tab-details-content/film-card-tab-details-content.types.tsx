// Types
import { TFilm } from "_Types/film";

export type TProps = {
  director: TFilm['director'];
  starring: TFilm['starring'];
  runTime: TFilm['runTime'];
  genre: TFilm['genre']
  released: TFilm['released']
}
