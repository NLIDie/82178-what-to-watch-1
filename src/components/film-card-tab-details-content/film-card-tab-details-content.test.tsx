import * as React from "react";
import {create} from "react-test-renderer";

// Components
import {FilmCardTabDetailsContent} from "./film-card-tab-details-content";

describe(`Components/FilmCardTabDetailsContent`, () => {
  it(`Correct renderer the component`, () => {
    const tree = create(
        <FilmCardTabDetailsContent
          director="director"
          genre="All genres"
          released={2014}
          runTime={95}
          starring={[`star1`, `star2`]}
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
