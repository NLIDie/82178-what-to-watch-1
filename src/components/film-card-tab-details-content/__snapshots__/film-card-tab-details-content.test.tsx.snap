// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/FilmCardTabDetailsContent Correct renderer the component 1`] = `
Array [
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        director
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        <span>
          star1
          <br />
        </span>
        <span>
          star2
          <br />
        </span>
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        95
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        All genres
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2014
      </span>
    </p>
  </div>,
]
`;
